src/rws_poco_client.cpp

```diff
@@ -334,20 +334,18 @@ POCOClient::POCOResult POCOClient::webSocketConnect(const std::string& uri,
   try
   {
     result.addHTTPRequestInfo(request);
+    
     {
       // We must have at least websocket_connect_mutext_.
       // If a connection already exists, we must also have websocket_use_mutex_.
       // If not, nobody should have the mutex anyway, so we should get it immediately.
       ScopedLock<Mutex> connect_lock(websocket_connect_mutex_);
       ScopedLock<Mutex> use_lock(websocket_use_mutex_);
+      
       p_websocket_ = new WebSocket(http_client_session_, request, response);
-
       p_websocket_->setReceiveTimeout(Poco::Timespan(timeout));
     }
 
-    p_websocket_ = new WebSocket(http_client_session_, request, response);
-    p_websocket_->setReceiveTimeout(Poco::Timespan(timeout));
-
```

Comment:
Were these lines added [here](https://github.com/ros-industrial/abb_librws/pull/58/files?utf8=%E2%9C%93&diff=unified&w=1#diff-6737ddfc5b7f1396c926faa2020236d4R348-R350)?

Can you explain why these were added?

Perhaps @de-vri-es can shed some light on that?

(note: I haven't followed the discussions around the previous PRs)

Reply:
Oops, my bad. I had an `if` checking for some condition deciding to take an extra mutex first, but that was a race condition. I fixed that by simply always taking the mutex, but in the process it appears I left the contents of the `else` branch.

Reply:
Oh, I messed up a merge conflict while rebasing.